pkg/appgw/internaltypes.go

```diff
@@ -70,7 +70,13 @@ func getResourceKey(namespace, name string) string {
 }
 
 func generateHTTPSettingsName(serviceName string, servicePort string, backendPortNo int32, ingress string) string {
```

Comment:
It looks like the logic here is to simply truncate the name to less than 80 characters. This is pretty error prone since now you can lead to collisions in names. We probably need a more novel strategy to generate names for each of the sub resource in order to avoid name collisions and still maintain a character limit of 80 characters.

Reply:
The last section of the deployment/ingress name is always unique isn't it? You could just pull the unique piece using a regex like `\-([a-z1-9]*)$`. In the test, the name would be `k8s-ag-ingress-j7sxh-8089-bp-8089-t8rnf`.

Reply:
I liked the approach suggested by @cwoolum.  I have implemented it and updated the PR.

Reply:
@cwoolum That's true for the pod names, but I'm not so sure about service and ingress. 

To illustrate this I added the log line here: https://github.com/Azure/application-gateway-kubernetes-ingress/pull/171 and cherry-picked the change in this PR.

For example - here is a set of test services I have on my AKS at the moment:
```
$ kubectl get service

NAME                      TYPE        CLUSTER-IP     EXTERNAL-IP   PORT(S)    AGE
frontend                  ClusterIP   10.2.61.105    <none>        80/TCP     10d
kubernetes                ClusterIP   10.2.0.1       <none>        443/TCP    10d
liveliness-test-service   ClusterIP   10.2.250.203   <none>        80/TCP     31h
redis-master              ClusterIP   10.2.66.56     <none>        6379/TCP   10d
```

And here are the ingress controllers:
```
$ kubectl get ingress

NAME                         HOSTS                 ADDRESS   PORTS     AGE
guestbook                    xx.mis.li,yy.mis.li             80, 443   6d15h
test-health-probes-ingress   xx.mis.li,yy.mis.li             80, 443   31h
```

And the log line indicates:  `I0511 23:31:17.313761   11190 backendhttpsettings.go:141] Created a new HTTP setting w/ name: k8s-ag-ingress-service-80-bp-8081-ingress`

The name constructed is `k8s-ag-ingress-service-80-bp-8081-ingress` -- we took `service` from `liveliness-test-service` and `ingress` from `test-health-probes-ingress`.